### PR TITLE
Refactor values

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/kubectl-gs/pull/678 and https://github.com/giantswarm/giantswarm/issues/20821

See `CHANGELOG.md` for a list of changes (there are lots of small ones).

### CODE REVIEW

**Comment by tfussell** [comment]
@giantswarm/team-honeybadger You might be interested as there are some implications for how we might migrate existing GitOps clusters when `values.yaml` structure changes.

**Comment by tfussell** [comment]
I tested this on `galaxy`.

File: helm/cluster-openstack/ci/values.yaml [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
+  useRootVolume: true
 
 nodePools:
-- name: default
-  class: default
-  replicas: 3
+- class: worker
+  failureDomain: az1
+  name: worker
+  replicas: 5
```
Comment (MarcelMue) [inline helm/cluster-openstack/ci/values.yaml:46]: Why this change?

File: helm/cluster-openstack/templates/_cluster_class.tpl [hunk 2]
```diff
@@ -16,7 +16,7 @@ spec:
       ref:
         apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
         kind: OpenStackMachineTemplate
-        name: {{ include "resource.default.name" . }}-{{ .Values.controlPlane.class }}
+        name: {{ include "resource.default.name" . }}-control-plane
```
Comment (MarcelMue) [inline helm/cluster-openstack/templates/_cluster_class.tpl:19]: Does a cluster upgrade like this work? Needs to be tested for sure.

**Review by MarcelMue — COMMENTED** [review]
Did you only test cluster creation or also cluster upgrades? Seems unclear to me.

File: helm/cluster-openstack/templates/_cluster_class.tpl (hunk 2 above)
Comment (tfussell) [reply helm/cluster-openstack/templates/_cluster_class.tpl:19]: If it doesn't, it's a big problem. I will test an upgrade from the previous version and report back.

**Comment by tfussell** [comment]
> Did you only test cluster creation or also cluster upgrades? Seems unclear to me.

Only creation. Upgrades are not a priority IMO as we explicitly don't support upgrades for alpha clusters. I will try though.

**Review by MarcelMue — APPROVED** [review]

File: helm/cluster-openstack/values.yaml [hunk 3]
```diff
@@ -1,41 +1,32 @@
...
-  replicas: 1  # Number of replicas in control plane. Should be an odd number.
-  availabilityZones: []
+  availabilityZones: []  # Unique availability zones for control plane nodes. If specified, should have one item or one for each replica (optional).
+  diskSize: 0  # Root volume disk size in GB. Ignored if bootFromVolume is false. Should be at least as large as the source image.
+  flavor: ""  # Project-dependent flavor of control plane machines.
+  image: ""  # Image name or root volume source UUID if bootFromVolume is true.
+  replicas: 0  # Number of replicas in control plane. Should be an odd number.
+  bootFromVolume: false  # If true, machines will use a persistent root volume instead of an ephemeral volume.
+
+nodeClasses: []  # Class definitions for worker node pools. In addition to a required "name", may include "diskSize", "flavor", "image", and "bootFromVolume" as defined for control plane and bastion above.
```
Comment (kopiczko) [inline helm/cluster-openstack/values.yaml:30]: I'd prefer the old approach of specifying fully one exemple entry. Or maybe two.

**Review by kopiczko — APPROVED** [review]
I like it. LGTM when tested.

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,10 +10,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add API server`certSANs` property to generate a certificate for the API FQDN.
+- Create separate `OpenStackMachineTemplate` for control plane machines.
+- Improve documentation for values in `values.yaml`.
+- Add `useRootVolume` option on bastion, control plane, and worker machine definitions to explicitly use non-ephemeral volumes.
```
Comment (kopiczko) [inline CHANGELOG.md:15]:
```suggestion
- Add `bootFromVolume ` option on bastion, control plane, and worker machine definitions to explicitly use non-ephemeral volumes.
```

File: helm/cluster-openstack/values.yaml (hunk 3 above)
Comment (tfussell) [reply helm/cluster-openstack/values.yaml:33]: I agree that it's nice to know what the structure looks like, but I don't like putting example values because it's installation specific. The user must provide values that work for their environment.

File: helm/cluster-openstack/templates/_cluster_class.tpl (hunk 2 above)
Comment (tfussell) [reply helm/cluster-openstack/templates/_cluster_class.tpl:19]: I'm sorry to report that cluster upgrades won't work because `OpenStackMachineTemplate` is immutable. This is not unique to this PR and would affect anything that changes the structure of a cluster. I have a PR to fix this using template rotation here https://github.com/giantswarm/cluster-openstack/pull/25. We're going to group this in with the OIDC changes and require that existing clusters are re-created to get this set of changes.
